pkg/synopsysctl/soperator_resources.go

```diff
@@ -274,3 +277,86 @@ func (specConfig *SOperatorSpecConfig) GetOperatorClusterRoleBinding() *horizonc
 
 	return synopsysOperatorClusterRoleBinding
 }
+
+// GetOperatorClusterRole creates a ClusterRole Horizon component for the Synopsys-Operaotor
+func (specConfig *SOperatorSpecConfig) GetOperatorClusterRole() *horizoncomponents.ClusterRole {
+	synopsysOperatorClusterRole := horizoncomponents.NewClusterRole(horizonapi.ClusterRoleConfig{
+		APIVersion: "rbac.authorization.k8s.io/v1beta1",
+		//ClusterName : "string,"
+		Name:      "synopsys-operator-admin",
+		Namespace: specConfig.Namespace,
+	})
+
+	synopsysOperatorClusterRole.AddPolicyRule(horizonapi.PolicyRuleConfig{
+		Verbs:           []string{"get", "list", "watch", "create", "update", "patch", "delete"},
+		APIGroups:       []string{"apiextensions.k8s.io"},
+		Resources:       []string{"customresourcedefinitions"},
+		ResourceNames:   []string{},
+		NonResourceURLs: []string{},
+	})
+
+	synopsysOperatorClusterRole.AddPolicyRule(horizonapi.PolicyRuleConfig{
+		Verbs:           []string{"get", "list", "watch", "create", "update", "patch", "delete"},
+		APIGroups:       []string{"rbac.authorization.k8s.io"},
+		Resources:       []string{"clusterrolebindings", "clusterroles"},
+		ResourceNames:   []string{},
+		NonResourceURLs: []string{},
+	})
+
+	synopsysOperatorClusterRole.AddPolicyRule(horizonapi.PolicyRuleConfig{
+		Verbs:           []string{"get", "list", "watch", "create", "update", "patch", "delete"},
+		APIGroups:       []string{""},
+		Resources:       []string{"namespaces", "configmaps", "persistentvolumeclaims", "services", "secrets", "replicationcontrollers", "deployments", "statefulsets", "serviceaccounts"},
```

Comment:
Not sure it's worth being this specific.  Unless we have to be more specific, I would probably make the resource list a "*".  Otherwise we may have to continually modify this as the operator gains more functionality/applications.

Reply:
But @rrati again it will look like cluster-admin right.. That's why we have resources name list..

Reply:
I changed the resource list to a "*"

Reply:
@msenmurugan In terms of control over the objects it will, but cluster-admin has more rights than just being able to do anything to cluster resources.  If we're this specific then we have to make sure to handle upgrade cases where we might add or remove resources we will need to control.  Not the end of the world (and more secure to do that), but it could complicate the upgrade path.  I don't really feel strongly either way.   I suppose until we know there is a complication we should air on the side of secure and list the resources we know we need right now.  So, I guess disregard my original comment and use a list of resources. :)

Reply:
I reverted the changes. And then moved "pods" from the rule below into this one.